packages/api/src/brevo-contact/brevo-contact-import.controller.ts

```diff
@@ -0,0 +1,53 @@
+import { CometValidationException, RequiredPermission } from "@comet/cms-api";
+import { EntityRepository } from "@mikro-orm/core";
+import { InjectRepository } from "@mikro-orm/nestjs";
+import { Body, Controller, Inject, Post, Type, UploadedFile, UseInterceptors } from "@nestjs/common";
+import { FileInterceptor } from "@nestjs/platform-express";
+import { TargetGroupInterface } from "src/target-group/entity/target-group-entity.factory";
+
+import { BrevoContactImportService } from "../brevo-contact/brevo-contact-import.service";
+import { ImporterErrorInterceptor } from "../brevo-contact/contact-importer-error.interceptor";
+import { BrevoModuleConfig } from "../config/brevo-module.config";
+import { BREVO_MODULE_CONFIG } from "../config/brevo-module.constants";
+import { EmailCampaignScopeInterface } from "../types";
+
+export function createBrevoContactImportController({ Scope }: { Scope: Type<EmailCampaignScopeInterface> }): Type<unknown> {
+    @Controller("brevo-contacts-csv")
+    class BrevoContactImportController {
+        constructor(
+            @Inject(BREVO_MODULE_CONFIG) private readonly config: BrevoModuleConfig,
+            @Inject(BrevoContactImportService) private readonly brevoContactImportService: BrevoContactImportService,
+            @InjectRepository("TargetGroup") private readonly targetGroupRepository: EntityRepository<TargetGroupInterface>,
+        ) {}
+
+        @Post("upload")
+        @UseInterceptors(
+            FileInterceptor("file", {
+                limits: { fileSize: 15 * 1024 * 1024 },
+                fileFilter: (_, file, cb) => {
+                    if (file.mimetype !== "text/csv") {
+                        return cb(new CometValidationException(`Unsupported mime type: ${file.mimetype}`), false);
+                    }
+
+                    return cb(null, true);
+                },
+            }),
+        )
+        @UseInterceptors(ImporterErrorInterceptor)
+        @RequiredPermission(["brevo-newsletter"], { skipScopeCheck: true })
+        async upload(@UploadedFile() file: Express.Multer.File, @Body("scope") scope: string, @Body("listIds") listIds?: string): Promise<void> {
+            const contentContent = file.buffer.toString("utf8");
+            const parsedScope = JSON.parse(scope) as EmailCampaignScopeInterface;
+
+            let parsedListIds = undefined;
+            if (listIds) parsedListIds = JSON.parse(listIds) as string[];
+
+            const targetGroupBrevoIds = (await this.targetGroupRepository.find({ id: { $in: parsedListIds } })).map(({ brevoId }) => brevoId);
+
+            const redirectUrl = this.config.brevo.resolveConfig(parsedScope).redirectUrlForImport;
+            await this.brevoContactImportService.importContactFromCsv(contentContent, parsedScope, redirectUrl, targetGroupBrevoIds);
```

Comment:
This could in theory take longer than 30s (default request timeout). @raphaelblum  Is this something we should take into consideration?

Reply:
Yes, I believe that will happen to us at some point so we should take it into consideration.

Probably not for the current use case, but when I look at the function, we send 1 request for the contact and then 1 for updating or creating a double opt in contact. If I calculate with a very fast api and a request only takes 25ms, that would be 30000 ms / 50 ms  = 600 Contacts. 

And we already had imports of 1000-5000 contacts.

Reply:
da6b4d313aaa2393481a41e0112c89ed34e8f728